web/src/components/upload/UploadsTableDeleteDialog.vue

```diff
@@ -0,0 +1,106 @@
+<script lang="ts" setup>
+import { UploadReport } from "@/types/uploads/UploadReport";
+import { useVModel } from "@vueuse/core";
+import { ref } from "vue";
+import axios from "axios";
+import { useSnackbar } from "../util/snackbar/useSnackbar";
+
+type Props = {
+  open: boolean;
+  report: UploadReport;
+  reports: UploadReport[];
+};
+const props = withDefaults(defineProps<Props>(), {
+  open: false,
+});
+const emit = defineEmits(["update:open", "update:reports"]);
+
+const open = useVModel(props, "open", emit);
+const reports = useVModel(props, "reports", emit);
+const loading = ref(false);
+const snackbar = useSnackbar();
+
+// Cancel the deletion.
+const cancel = (): void => {
+  open.value = false;
+};
+
+// Confirm the deletion.
+const confirm = async (): Promise<void> => {
```

Comment:
I think functions like this are maybe better collected in a `report.store.ts`. Deleting a report could be something we also want to do from other places in the application as well.